### ISSUE TITLE
Fractional Scaling

### ISSUE THREAD

**Issue by Thaodan** [masm11/emacs#92]
Is that supported? I tried to set it up with:
```
export GDK_DPI_SCALE=0.5
export GDK_SCALE=2
```
However I can't make it work.
My DE is Plasma 20.0.5.

**Comment by fejfighter** [comment]
I don't think we explicitly do support GDK_DPI_SCALE, it's listed as a x specific in gtk3 documentation

is your plasma session running on X or wayland? apparently neither work on wayland, but are expected to have some effect on X

**Comment by masm11** [comment]
https://developer.gnome.org/gtk3/stable/gtk-x11.html#x11-envar

I didn't implement X11-specific feature explicitly.

**Comment by Thaodan** [comment]
My session is on X11.

Anyway fractional scaling support would be great. 2.0 scaling is to be for me on 4k and 27".

**Comment by masm11** [comment]
OK, I'll consider whether it can be implemented.
Does X-build emacs support it?

**Comment by Thaodan** [comment]
> OK, I'll consider whether it can be implemented.
> Does X-build emacs support it?
Not really but you can use the XRessource `Xft.dpi` to simulate it.

**Comment by masm11** [comment]
I don't use Xft.dpi because I can't get it in pure-gtk way.
Instead, I'll use GDK_DPI_SCALE.

**Comment by Thaodan** [comment]
> I don't use Xft.dpi because I can't get it in pure-gtk way.
Sure but you asked about the GTK build that had X11 bits next to GTK.

**Comment by masm11** [comment]
> Sure but you asked about the GTK build that had X11 bits next to GTK.

Sorry, I misunderstood your words.

Thanks.  I'll implement in pgtk while referring to how to use Xft.dpi in X-build.

**Comment by masm11** [comment]
I tried to support GDK_DPI_SCALE, and push it to feature/pgtk of https://github.com/masm11/emacs.
Could you try it?

**Comment by masm11** [comment]
Did you try it?

**Comment by oabdullah1** [comment]
For what is worth I tried the latest code from emacs-mirror:feature/pgtk and it supports fractional scaling under Wayland already without this patch. Maybe it's only required for X11?

**Comment by masm11** [comment]
@oabdullah1 Yes, it's only for X11.

**Comment by Thaodan** [comment]
Hey sorry for the late reply.

It works for the ui however  fonts are not scaled up (compare the fonts on the ui in grey vs. the text black/white).
![Screenshot_20210419_023904](https://user-images.githubusercontent.com/1747359/115165854-9fc75700-a0b8-11eb-9aea-87b6a71cee00.png)

**Comment by masm11** [comment]
It works on my GNOME(Xorg).
Did you try feature/pgtk on https://github.com/masm11/emacs, not savannah?  savannah's one doesn't contains this support.

![scaled](https://user-images.githubusercontent.com/7226237/115258840-4f113600-a16c-11eb-8771-7952d465c338.png)

I'll try it on Plasma later.

**Comment by masm11** [comment]
It works on Plasma 5.21.4.

![works_on_plasma](https://user-images.githubusercontent.com/7226237/115398863-8d692c80-a222-11eb-8ba0-7281fcbf37e8.png)
